MediaProcessor/src/AudioProcessor.cpp

```diff
@@ -15,19 +15,40 @@ AudioProcessor::AudioProcessor(const std::string &inputVideoPath,
                                const std::string &outputAudioPath)
     : m_inputVideoPath(inputVideoPath),
       m_outputAudioPath(outputAudioPath),
-      m_numChunks(DEFAULT_NUM_CHUNKS),
+      m_numChunks(determineNumChunks()),
       m_overlapDuration(DEFAULT_OVERLAP_DURATION) {
     m_outputDir = std::filesystem::path(outputAudioPath).parent_path().string();
     m_chunksDir = m_outputDir + "/chunks";
     m_processedChunksDir = m_outputDir + "/processed_chunks";
 
+    std::cout << "INFO: using " << std::to_string(m_numChunks) << " Threads." << std::endl;
+
     /*
      * TODO: 
      * * numChunks (numThreads) should be configurable, as well as the overlap (which seems much less critical?)
      * * Need an ffmpeg utils to establish an API for extaction and probing
      */
 }
 
+int AudioProcessor::determineNumChunks() const {
```

Comment:
Please move this method to under `ConfigManager` and rename it to `getOptimalThreadCount`.
This new getter should call a private method `determineNumThreads`, where we determine this based off of hardware information. 

For this hardware information, use add a new `HardwareUtils` with a `getHardwareThreadCount()`. This will simply return the info via the `std::thread::hardware_concurrency()`.

Roughly:
```cpp
// in ConfigManager
unsigned  getOptimalThreadCount() {
    unsigned int configNumThreads = ... // get from config
    unsigned int hardwareNumThreads = MediaProcessor::HardwareUtils::getHardwareThreadCount();

    return determineNumThreads(configNumThreads, hardwareNumThreads);
}
```

This will help avoid coupling loosely related classes together.